### PR TITLE
feat: Enable fully coupled mode for CompositionalMultiphaseReservoirPoromechanics and SinglePhaseReservoirPoromechanics

### PR DESCRIPTION
**Pull request by paveltomin**
should go after https://github.com/GEOS-DEV/GEOS/pull/3422

- works equivalent to `MultiphasePoromechanicsReservoir` and `SinglePhasePoromechanicsReservoir`, just more convenient to switch between FIM and SEQ within the same solver
- can use the same MGR recipe (order of block is actually the same: mechanics, flow, wells)

### CODE REVIEW

**Comment by paveltomin** [comment]
need review please

**Comment by CusiniM** [comment]
I am lost. What are these? What s the difference between these and  `MultiphasePoromechanicsReservoir` and `SinglePhasePoromechanicsReservoir` ?

**Comment by paveltomin** [comment]
> I am lost. What are these? What s the difference between these and `MultiphasePoromechanicsReservoir` and `SinglePhasePoromechanicsReservoir` ?

`CompositionalMultiphaseReservoirPoromechanics` and `SinglePhaseReservoirPoromechanics` were created for sequential since `MultiphasePoromechanicsReservoir` and `SinglePhasePoromechanicsReservoir` are designed differently: they are flow/mech + wells so when one runs them in sequential they will do flow/mech solve and wells solve iteratively.
`CompositionalMultiphaseReservoirPoromechanics` and `SinglePhaseReservoirPoromechanics` were not be able to run in FIM mode which is addressed in this PR.
I know this is confusing and ultimately we should clean this up.

**Review by jhuang2601 — APPROVED** [review]

File: src/coreComponents/physicsSolvers/multiphysics/MultiphasePoromechanics.cpp [hunk 1]
```diff
@@ -296,6 +335,33 @@ void MultiphasePoromechanics<>::setMGRStrategy()
...
+  }
+  else
+  {
+    // add Reservoir
+    linearSolverParameters.mgr.strategy = LinearSolverParameters::MGR::StrategyType::multiphasePoromechanicsReservoirFVM;
+  }
+  GEOS_LOG_LEVEL_RANK_0( 1, GEOS_FMT( "{}: MGR strategy set to {}", getName(),
+                                      EnumStrings< LinearSolverParameters::MGR::StrategyType >::toString( linearSolverParameters.mgr.strategy )));
+  // TODO check if needed
+  //linearSolverParameters.mgr.separateComponents = true;
```
Comment (victorapm) [inline src/coreComponents/physicsSolvers/multiphysics/MultiphasePoromechanics.cpp]: This should be needed, but next line can be removed

File: src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanics.cpp [hunk 2]
```diff
@@ -139,6 +132,33 @@ void SinglePhasePoromechanics<>::setMGRStrategy()
...
+  }
+  else
+  {
+    // add Reservoir
+    linearSolverParameters.mgr.strategy = LinearSolverParameters::MGR::StrategyType::singlePhasePoromechanicsReservoirFVM;
+  }
+  GEOS_LOG_LEVEL_RANK_0( 1, GEOS_FMT( "{}: MGR strategy set to {}", this->getName(),
+                                      EnumStrings< LinearSolverParameters::MGR::StrategyType >::toString( linearSolverParameters.mgr.strategy )));
+  // TODO check if needed
+  //linearSolverParameters.mgr.separateComponents = true;
```
Comment (victorapm) [inline src/coreComponents/physicsSolvers/multiphysics/SinglePhasePoromechanics.cpp]: Same comment as before

**Review by victorapm — APPROVED** [review]
Thanks!

**Review by CusiniM — COMMENTED** [review]
Shouldn't we remove the old solvers though? I would only keep this one with the two options in the same solver.

**Review by CusiniM — APPROVED** [review]

**Review by CusiniM — CHANGES_REQUESTED** [review]
Actually I would strongly prefer getting rid of `MultiphasePoromechanicsReservoir` and `SinglePhasePoromechanicsReservoir`

**Comment by paveltomin** [comment]
> Actually I would strongly prefer getting rid of `MultiphasePoromechanicsReservoir` and `SinglePhasePoromechanicsReservoir`

One reason they can be useful is to perform an isolated well solve to precondition the nonlinear system, that's why I am hesitant to remove them

**Review by CusiniM — APPROVED** [review]

**Pull request merged**